### PR TITLE
Base: add support for querying user home directory on Windows

### PR DESCRIPTION
**Pull request by compnerd**
The user profile directory is not possible to access without the user
token for the `GetUserProfileDirectoryW`.  Instead, we can iterate the
users matching a particular filter (e.g. local users).  This allows us
to at least query some user home directory.  Unfortunately, this would
disallow querying of users from a domain controller.  This resolves an
immediate issue of SPM tests depending on the ability to query profile
information for a user.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @parkera

**Comment by compnerd** [comment]
@swift-ci please test

File: CoreFoundation/Base.subproj/CFPlatform.c [hunk 1]
```diff
@@ -577,15 +578,42 @@ CF_EXPORT CFURLRef CFCopyHomeDirectoryURLForUser(CFStringRef uName) {
...
+                            &dwToken);
+      if (nStatus == NERR_Success || nStatus == ERROR_MORE_DATA) {
+        for (unsigned uiEntry = 0; !home && uiEntry < dwEntriesRead; ++uiEntry) {
+          CFStringRef name =
+              CFStringCreateWithCStringNoCopy(kCFAllocatorSystemDefault,
+                                              pBuffer[uiEntry].usri1_name,
+                                              kCFStringEncodingUTF16,
+                                              kCFAllocatorNull);
+
+          if (CFEqual(uName, name))
```
Comment (parkera) [inline CoreFoundation/Base.subproj/CFPlatform.c:602]: Code style nit, but we require `{ }` if not a 1-line `if`. Same applies on line 582.

**Comment by compnerd** [comment]
@swift-ci please test

**Review by parkera — APPROVED** [review]

**Pull request merged**